### ISSUE TITLE
Reicast OIT - a question when it comes to performance (THPS2 especially)

### ISSUE THREAD

**Issue by Gagert** [libretro/flycast#51]
I just recently stumbled upon the OIT reicast libretro core, so I had to give it a go. I had just unlocked the Philadelphia stage in Tony Hawk's Pro Skater 2 using the newer build of the regular reicast core, and I noticed that using the regular reicast core it weren't any graphic drawn while doing manuals or after filling up the special meter. Other than that the game only had minor audio issues (sometimes while crashing through the windows in Hangar there weren't any window-breaking sounds) and it ran smooth as butter.

Now back to the OIT reicast libretro core. Now the graphics where drawn both at the special meter and while doing manuals, but oh boy the performance certainly took a big hit in my case. I just read the blogpost on the libretro home page regarding the newly added OIT reicast core, that the core itself fixed a lot of graphical issues while also being more demanding performance-wise compared to the regular reicast core.

Playing Tony Hawk's Pro Skater 2 on the OIT core makes the game struggle quite a bit on my current hardware, and the music goes slow-motion at certain points (especially at the Philadelphia stage). My current rig has an i7 4790K, GTX 1080 and 16 gb ram, so it's still a quite decent gaming PC in my personal opinion.

I'm just wondering if the core is supposed to be that demanding to a PC such as mine (I'm using an internal resolution at 1280x960), or if it actually might be a bug/resource leak that's currently making the core that demanding? I have tried different settings using the OIT core, but without any success getting a stable 60 fps in THPS2 on my rig.

Maybe I forgot turning a setting on or something in RetroArch? Have any of you tested THPS2 in the reicast OIT core, and how was the performance? Also, what are your system specs?

**Comment by inactive123** [comment]
We will have to wait for @flyinghead to come back from vacation before we can discuss performance-related questions with regards to the OIT renderer. 

For now I'd suggest you lower the resolution since it seems to matter a lot with the OIT renderer.

**Comment by flyinghead** [comment]
For some reason TPHS2 hangs as soon as I start a game (career, free, ..) This only happens with the libretro cores: my branch runs fine. I've tried both the US and UK versions with libretro to no avail.

Are you using special settings to run this game?

On my OIT branch the first area (Mullet Falls) runs fine at 50 FPS with no frame skipped. I have an i7-6900K with 32 GB and a GTX 1080.

**Comment by Gagert** [comment]
I'm using RetroArch 1.6.9 (I haven't updated yet), I suppose I'm using somewhat default settings and I'm launching the ReiCast OIT core via a batch-file I made in Windows 10 (tony hawk 2.bat). Inside the batch file I wrote:

retroarch.exe -f -L "D:\Emulators\RetroArch\RetroArch 1.6.9\cores\reicast_oit_libretro.dll" "D:\Sega Dreamcast iso\Tony Hawk's Pro Skater 2\Tony Hawk's Pro Skater 2.cdi"

I ran into some issues trying to run the game via the RetroArch GUI itself using the OIT core.

Edit: I wrote this reply way to quick; I could not initially make the ReiCast OIT core launch Dreamcast games within RetroArch 1.6.9, because after opening a game I could only choose the standard ReiCast core (not the OIT one). By launching a game in the ReiCast OIT core via a batch-file, I could then re-launch the game later on in the ReiCast core using the History/Log section in the GUI.
